Add forgotten async function to the list

atrosinenko opened the pull request:
Add forgotten `emscripten_yield` function to the list of async functions in emterpretify.py just like it was done for Asyncify implementation, so `EMTERPRETIFY_ADVISE` know about it.

I didn't done thorough research, but I have not added `emscripten_coroutine_next` supposing that if it was not required for Asyncify, then hope it is not required for Emterpreter too. :)

kripken commented:
Thanks! This looks correct to me.

Please add a test for this. We have a few emterpreter-async tests in `browser`, or maybe the `other` test for emterpretify-advise is a better place for it - maybe we can add to one of those.

atrosinenko commented:
I have implemented a test, but placed it to `test_core.py`, so it is located near existing coroutine tests. I have adjusted previously written coroutine test to emterpret just three required functions (and test that it works with such setting) and tested that they are those functions that are advised by `EMTERPRETIFY_ADVISE`. I suppose these two tests should be in one place, so it would be simpler to update these two function lists synchronously (and as far as I remember, I placed first emterpretify coroutine test to `test_core`, because there already was a test for asyncify coroutines).

kripken commented on tests/test_core.py:
```diff
@@ -7315,59 +7315,29 @@ def test_async_abort(self):
   def test_coroutine_asyncify(self):
     self.do_test_coroutine({'ASYNCIFY': 1})
 
+  @no_wasm_backend('EMTERPRETIFY causes JSOptimizer to run, which is '
```
Thanks!

The first test makes sense to put in this file, and looks good. The second one, `test_coroutine_emterpretify_advise`, should be in `other` (alongside the existing `EMTERPRETIFY_ADVISE` test perhaps), since it compiles manually itself - so it doesn't need to be run in each compilation mode separately, which is what tests in test_core get.

kripken commented:
Perfect, thanks!

kripken commented:
@atrosinenko this broke `asm1.test_coroutine_emterpretify_async`. specifically it broke asm.js validation, which is not tested on the bots, so we didn't notice this immediately.

kripken commented:
Proposed fix in #6479

kripken commented:
The breakage was due to the new test here, which uncovered an existing bug that that PR fixes.

atrosinenko commented:
@kripken Does #6479 fix this issue completely or is some action expected from me?

kripken commented:
@atrosinenko It should be fixed by that PR, nothing else needed - thanks for checking.
